### ISSUE TITLE
syncthing: Add support for remoteIgnoredDevice

### ISSUE THREAD

**Issue by DamienCassou** [nix-community/home-manager#6136]
syncthing supports a global `remoteIgnoredDevice` option which allows for ignoring connection requests from some computers:

```xml
<configuration version="37">
    ...
    <remoteIgnoredDevice time="2024-11-24T11:35:15.551Z" id="MH4OL5B-..." name="someone-else" address="some.ip.add.ress"></remoteIgnoredDevice>
</configuration>
```

https://docs.syncthing.net/users/config.html#config-option-configuration.remoteignoreddevice

Is that something it makes sense to add?

/cc @rycee @pitkling @karaolidis

**Comment by rycee** [comment]
If I understand things correctly, this should already be possible using [`services.syncthing.settings`](https://nix-community.github.io/home-manager/options.xhtml#opt-services.syncthing.settings). Something like:

``` nix
services.syncthing.settings = {
  remoteIgnoredDevices = [
    {
      time = "2024-11-24T11:35:15.551Z";
      deviceID = "MH4OL5B-...";
      name = "someone-else";
      address = "some.ip.add.ress";
    }
  ];
};
```

perhaps? I'm not certain as I don't use declarative syncthing configurations myself.

**Comment by DamienCassou** [comment]
This seems to work, thank you! Should we close the issue? Add something to the documentation?

**Comment by aguirre-matteo** [comment]
> This seems to work, thank you! Should we close the issue? Add something to the documentation?

@DamienCassou Yep, close it. It's a convention to use the `<any>.settings` for configuring all/most options.
